src/Microsoft.OData.Core/Uri/NodeToStringBuilder.cs

```diff
@@ -470,7 +471,245 @@ internal String TranslateNode(QueryNode node)
+            }
+            else
+            {
+                return this.TranslateNode(node);
+            }
+        }
+
+        /// <summary>
+        /// Bind $it to the <see cref="BinaryOperatorNode"/> translated string.
+        /// </summary>
+        /// <param name="node">node to bind.</param>
+        /// <param name="filterClauseRangeVariable">The <see cref="FilterClause"/> range variable.</param>
+        /// <returns>The translated string with $it binding.</returns>
+        private string BindBinaryOperatorNode(BinaryOperatorNode node, ResourceRangeVariable filterClauseRangeVariable)
+        {
+            string translatedNode = this.TranslateNode(node);
+            ResourceRangeVariableReferenceNode leftRangeVariableNode = GetResourceRangeVariableReferenceNode(node.Left);
+            ResourceRangeVariableReferenceNode rightRangeVariableNode = GetResourceRangeVariableReferenceNode(node.Right);
+
+            BinaryOperatorKind binaryNodeKind = node.OperatorKind;
+
+            string binaryOperator = this.BinaryOperatorNodeToString(binaryNodeKind);
+            string[] binarySeparator = { binaryOperator };
+            string[] substrings = translatedNode.Trim().Split(binarySeparator, StringSplitOptions.RemoveEmptyEntries);
+            string leftBinary = substrings[0].Trim();
+            string rightBinary = substrings[1].Trim();
+
+            if (leftRangeVariableNode != null && IsDifferentSource(filterClauseRangeVariable, leftRangeVariableNode))
+            {
+                leftBinary = ExpressionConstants.It + ExpressionConstants.SymbolForwardSlash + leftBinary;
+            }
+
+            if (rightRangeVariableNode != null && IsDifferentSource(filterClauseRangeVariable, rightRangeVariableNode))
+            {
+                rightBinary = ExpressionConstants.It + ExpressionConstants.SymbolForwardSlash + rightBinary;
+            }
+
+            return leftBinary + ' ' + binaryOperator + ' ' + rightBinary;
+        }
+
+        /// <summary>
+        /// Bind $it to the <see cref="InNode"/> translated string.
+        /// </summary>
+        /// <param name="node">node to bind.</param>
+        /// <param name="filterClauseRangeVariable">The <see cref="FilterClause"/> range variable.</param>
+        /// <returns>The translated string with $it binding.</returns>
+        private string BindInNode(InNode node, ResourceRangeVariable filterClauseRangeVariable)
+        {
+            string translatedNode = this.TranslateNode(node);
+            ResourceRangeVariableReferenceNode leftRangeVariableNode = GetResourceRangeVariableReferenceNode(node.Left);
+
+            string inOperator = ExpressionConstants.KeywordIn;
+            string[] inSeparator = { inOperator };
+            string[] substrings = translatedNode.Trim().Split(inSeparator, StringSplitOptions.RemoveEmptyEntries);
+            string leftIn = substrings[0];
+            string rightIn = substrings[1];

```

Comment:
This logic only needs to execute if the sources are different, so it would be better to move it within your if statement.